src/TestFramework/Factory.php

```diff
@@ -116,10 +121,23 @@ public function create(string $adapterName, bool $skipCoverage): AbstractTestFra
             );
         }
 
+        if ($adapterName === TestFrameworkTypes::CODECEPTION) {
+            $codeceptionConfigPath = $this->configLocator->locate(TestFrameworkTypes::CODECEPTION);
+            $codeceptionConfigContent = file_get_contents($codeceptionConfigPath);
```

Comment:
Please add a `\assert(is_string($codeceptionConfigContent));`

Reply:
Have you any objections to a general clean up? I feel like the entire function can likely be refactored such that modifications to the file would not be necessary if a new framework were to be added. Not a really big issue, but something I was considering.

Reply:
I'm all for general clean up. If you do so please move the clean up part to another PR. It allows us  to get the clean up part into the code base earlier, and makes reviewing this PR easier.